### PR TITLE
MAYA-114709 As a user, I'd like to see a waiting cursor when pulling …

### PR DESCRIPTION
**Pull request by boudrey**
…or duplicating to Maya data

### CODE REVIEW

**Review by seando-adsk — APPROVED** [review]
The changes are good, but what do you think about creating a little helper class where constuctor/destructor toggle the cursor?

**Comment by boudrey** [comment]
Still would not guaranty that cursor would actually be restore. The real fix would be to publish TwaitCursor to OpenMaya and create the Python binding for it. Yes, I can wrap it in a function for now.

File: lib/mayaUsd/ufe/UsdContextOps.cpp [hunk 1]
```diff
@@ -160,6 +160,25 @@ static const std::vector<std::string> kSchemaNiceNames = {
 // clang-format on
 #endif
 
+//! \brief Change the cursor to wait state when start() is called and restore it on destruction.
+struct MWaitCursor
+{
+    bool _started { false };
+
+    MWaitCursor() = default;
+    void start()
```
Comment (ppt-adsk) [inline lib/mayaUsd/ufe/UsdContextOps.cpp:169]: Why do we need the start() method?  Can't we just start in the constructor, and get rid of the _started data member?  Also not sure about the "M" prefix in the name.

File: plugin/adsk/scripts/USDMenuProc.mel [hunk 2]
```diff
@@ -47,9 +47,9 @@ global proc USDMenuProc(string $parent, string $obj)
             string $ls = tolower($s);
             if (`match $ls "true|yes|on|1"` !="")
             {
-                menuItem -label "Edit As Maya Data" -image "edit_as_Maya.png" -command ("mayaUsdEditAsMaya \"" + $obj + "\"");
+                menuItem -label "Edit As Maya Data" -image "edit_as_Maya.png" -command ("{waitCursor -state 1; mayaUsdEditAsMaya \"" + $obj + "\"; waitCursor -state 0;}");
```
Comment (ppt-adsk) [inline plugin/adsk/scripts/USDMenuProc.mel:50]: Hey!  Maybe you can do a guard object in MEL!

File: lib/mayaUsd/ufe/UsdContextOps.cpp (hunk 1 above)
Comment (boudrey) [reply lib/mayaUsd/ufe/UsdContextOps.cpp:169]: This is how this class should look like if it was part of OpenMaya, and hopefully one day.

File: plugin/adsk/scripts/USDMenuProc.mel (hunk 2 above)
Comment (boudrey) [reply plugin/adsk/scripts/USDMenuProc.mel:50]: Please let me know if you know how to do it.

Comment (ppt-adsk) [reply plugin/adsk/scripts/USDMenuProc.mel:50]: I would be very surprised if this would be possible :)  MEL is what it is.

**Review by ppt-adsk — APPROVED** [review]

File: lib/mayaUsd/ufe/UsdContextOps.cpp [hunk 3]
```diff
@@ -160,6 +160,13 @@ static const std::vector<std::string> kSchemaNiceNames = {
 // clang-format on
 #endif
 
+//! \brief Change the cursor to wait state when start() is called and restore it on destruction.
```
Comment (seando-adsk) [inline lib/mayaUsd/ufe/UsdContextOps.cpp:163]: Minor - there is no more start method.

**Review by ppt-adsk — APPROVED** [review]

**Review by seando-adsk — APPROVED** [review]

**Pull request merged**